From 2b606fcd434b993293cb334f23bf1b6d006f24b1 Mon Sep 17 00:00:00 2001
From: Daniel Borkmann <dborkman@redhat.com>
Date: Tue, 22 Jan 2013 13:28:08 +0100
Subject: [PATCH 38/53] test_1_to_1_sockopt: fix deprecated
 SO_RCVBUF/SO_SNDBUF testcase

Actually, it is not guaranteed, that the rcv/snd buffer size, we send
is really set by the kernel. For small values, the SOCK_MIN_SNDBUF and
SOCK_MIN_RCVBUF sizes in include/net/sock.h are used instead. This was
the sole reason why this test, whose implementation was in 2005, failed.
Just change it to something bigger is not critical, and we're good to go.

$ ./test_1_to_1_sockopt
test_1_to_1_sockopt.c  1 PASS : setsockopt() with a bad socket descriptor - EBADF
test_1_to_1_sockopt.c  2 PASS : setsockopt() with an invalid socket - ENOTSOCK
test_1_to_1_sockopt.c  3 PASS : setsockopt() with an invalid level - ENOPROTOOPT
test_1_to_1_sockopt.c  4 PASS : setsockopt() with invalid option buffer - EFAULT
test_1_to_1_sockopt.c  5 PASS : setsockopt() with invalid option name - EOPNOTSUPP
test_1_to_1_sockopt.c  6 PASS : getsockopt() with a bad socket descriptor - EBADF
test_1_to_1_sockopt.c  7 PASS : getsockopt() with an invalid socket - ENOTSOCK
test_1_to_1_sockopt.c  8 PASS : getsockopt() with invalid option buffer - EFAULT
test_1_to_1_sockopt.c  9 PASS : getsockopt() with invalid option name - EOPNOTSUPP
test_1_to_1_sockopt.c 10 PASS : getsockopt() SCTP_INITMSG - SUCCESS
test_1_to_1_sockopt.c 11 PASS : setsockopt() SCTP_INITMSG - SUCCESS
test_1_to_1_sockopt.c 12 PASS : setsockopt() SO_LINGER - SUCCESS
test_1_to_1_sockopt.c 13 PASS : getsockopt() SO_LINGER - SUCCESS
test_1_to_1_sockopt.c 14 PASS : getsockopt() SO_RCVBUF - SUCCESS
test_1_to_1_sockopt.c 15 PASS : getsockopt() SCTP_STATUS - SUCCESS
test_1_to_1_sockopt.c 16 PASS : setsockopt() SO_RCVBUF - SUCCESS
test_1_to_1_sockopt.c 17 PASS : setsockopt() SO_SNDBUF - SUCCESS
test_1_to_1_sockopt.c 18 PASS : getsockopt() SO_SNDBUF - SUCCESS
test_1_to_1_sockopt.c 19 PASS : getsockopt() SCTP_PRIMARY_ADDR - SUCCESS
test_1_to_1_sockopt.c 20 PASS : setsockopt() SCTP_PRIMARY_ADDR - SUCCESS
test_1_to_1_sockopt.c 21 PASS : getsockopt() SCTP_ASSOCINFO - SUCCESS
test_1_to_1_sockopt.c 22 PASS : setsockopt() SCTP_ASSOCINFO - SUCCESS

Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
---
 src/func_tests/test_1_to_1_sockopt.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/func_tests/test_1_to_1_sockopt.c b/src/func_tests/test_1_to_1_sockopt.c
index c3a4833..37aac1d 100644
--- a/src/func_tests/test_1_to_1_sockopt.c
+++ b/src/func_tests/test_1_to_1_sockopt.c
@@ -301,7 +301,7 @@ main(void)
 	/* Reducing the SO_RCVBUF value using setsockopt() */
 	/*Minimum value is 128 and hence I am using it*/
 	len = sizeof(int);
-	rcvbuf_val_set = 128;
+	rcvbuf_val_set = 2048;
 	/* TEST16: Test case for setsockopt SO_RCVBUF */
 	error = setsockopt(sk2, SOL_SOCKET, SO_RCVBUF, &rcvbuf_val_set, len);
 	if (error < 0)
@@ -320,7 +320,7 @@ main(void)
 			 "got value differs Set Value=%d Get Value=%d",
 			 (2*rcvbuf_val_set), rcvbuf_val_get);
 
-	sndbuf_val_set=1024;
+	sndbuf_val_set = 2048;
 	/* TEST17: Test case for setsockopt SO_SNDBUF */
 	error = setsockopt(sk2, SOL_SOCKET, SO_SNDBUF, &sndbuf_val_set, len);
 	if (error < 0)
-- 
1.7.11.7

